packages/node/src/indexer/BlockedQueue.ts

```diff
@@ -22,10 +22,31 @@ export class BlockedQueue<T> {
     this._queue.push(item);
   }
 
+  putAll(items: T[]): void {
+    if (this._queue.length >= this._maxSize) {
+      throw new Error('BlockedQueue exceed max size');
+    }
+    this._queue = this._queue.concat(items);
+  }
+
   async take(): Promise<T> {
     while (!this.size) {
       await delay(0.1);
     }
     return this._queue.shift();
   }
+  async takeAll(max?: number): Promise<T[]> {
+    while (!this.size) {
+      await delay(0.1);
+    }
+    let result;
+    if (max) {
+      result = this._queue.slice(0, max);
+      this._queue = this._queue.slice(max);
+    } else {
+      result = [...this._queue];
```

Comment:
return `this._queue directly` seems fine to me, since it is replaced with an empty array right after.